onnxruntime/python/onnxruntime_pybind_mlvalue.cc

```diff
@@ -181,6 +185,26 @@ std::unique_ptr<IDataTransfer> GetGPUDataTransfer() {
 
 #endif
 
+#ifdef USE_DML
+AllocatorPtr GetDmlAllocator(OrtDevice::DeviceId id) {
+  // Current approach is not thread-safe, but there are some bigger infra pieces to put together in order to make
+  // multi-threaded DML allocation work we need to maintain a per-thread DML allocator
+
+  // We are leaking this map so we do not accidentally destroy DML Allocator instance
```

Comment:
>We are leaking this map [](http://example.com/codeflow?start=5&length=23)

What do we need to do to properly clean up? Leaking as a strategy assumes people only use a library for the lifetime of a process, rather than load/unload them, which larger products like Adobe might do.

Reply:
I think the whole binding mechanism would need to be rethought honestly. This is inheriting the same startegy as CUDA, which also leaks the map. This is because binding doesn't take any context as parameters, so we cannot store this information anywhere.

Since this is only a python API, I'm not too worried about loading/unloading.